src/contract/deploy/impl/DefaultCreateContract.ts

```diff
@@ -85,9 +91,34 @@ export class DefaultCreateContract implements CreateContract {
     contractTX.addTag(SmartWeaveTags.SDK, 'RedStone');
     if (data) {
       contractTX.addTag(SmartWeaveTags.CONTENT_TYPE, data['Content-Type']);
-      contractTX.addTag(SmartWeaveTags.INIT_STATE, initState);
+      contractTX.addTag(SmartWeaveTags.INIT_STATE_FORMAT, stateFormat);
+      contractTX.addTag(
+        SmartWeaveTags.INIT_STATE,
+        typeof initState === 'string' ? initState : new TextDecoder().decode(initState)
```

Comment:
I haven't tested it outside of Arlocal yet and I'm not sure how if Mainnet would appreciate a transaction having a binary tag value? Anyway it doesn't really seem right, it should probably be encoded in base64... What do you think? And if base64, should we use base64url or plain base64?

Reply:
I believe Arweave requires all tags to be encoded in Base64Url - https://github.com/ArweaveTeam/arweave-js/blob/master/src/common/lib/transaction.ts#L125

Reply:
Oh that's true, completely forgot about that! So it's handled for us then, perfect!